permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/ComponentInfoFlagsFlutterApiImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.baseflow.permissionhandler;
+
+import android.content.pm.PackageManager.ComponentInfoFlags;
+
+import androidx.annotation.NonNull;
+
+import com.baseflow.instancemanager.InstanceManager;
+import com.baseflow.permissionhandler.PermissionHandlerPigeon.ComponentInfoFlagsFlutterApi;
+
+import java.util.UUID;
+
+import io.flutter.plugin.common.BinaryMessenger;
+
+/**
+ * Flutter API implementation for `ComponentInfoFlags`.
+ *
+ * <p>This class may handle adding native instances that are attached to a Dart instance or passing
+ * arguments of callbacks methods to a Dart instance.
+ */
+public class ComponentInfoFlagsFlutterApiImpl {
+    // To ease adding additional methods, this value is added prematurely.
+    @SuppressWarnings({"unused", "FieldCanBeLocal"})
+    private final BinaryMessenger binaryMessenger;
```

Comment:
Thinking a bit ahead, I don't see we would need the `binaryMessenger` anywhere else except for the constructor of the class. The `binaryMessenger` should only be used to pass into Flutter API implementation classes and these are always created in the constructor. Therefore I feel we can remove this field (and also in already existing code).